WolvenKit.App/ViewModels/Tools/AssetBrowserViewModel.cs

```diff
@@ -505,138 +505,116 @@ public async Task PerformSearch(string query)
         {
             Unknown,
             Include,
             Exclude,
-            Hash,
-            Verbatim
         }
 
         private readonly record struct Term(TermType Type, string Pattern, string NegationPattern);
 
+        // Refinement types
         private interface SearchRefinement { }
         private readonly record struct PatternRefinement(Term[] Terms) : SearchRefinement;
         private readonly record struct HashRefinement(ulong Hash) : SearchRefinement;
+        private readonly record struct RegexRefinement(Regex Regex) : SearchRefinement;
+        private readonly record struct VerbatimRefinement(string Verbatim) : SearchRefinement;
+
+        // Refinement type matchers
+        private static readonly Regex RefinementSeparator = new("\\s+>\\s+", RegexpOpts, RegexpSafetyTimeout);
+        private static readonly Regex IsHashRefinement = new("^hash:(?<numbers>\\d+)$", RegexpOpts, RegexpSafetyTimeout);
+        private static readonly Regex IsRegexRefinement = new("^r:(?<pattern>.*)$", RegexpOpts, RegexpSafetyTimeout);
+        private static readonly Regex IsVerbatimRefinement = new("^@(?<verbatim>.*)$", RegexpOpts, RegexpSafetyTimeout);
 
         private readonly record struct CyberSearch(Func<IGameFile, bool> Match, SearchRefinement SourceRefinement);
 
         // Term to refinement pattern conversion regexps
 
-        private static readonly RegexOptions RegexpOpts = RegexOptions.Compiled | RegexOptions.IgnoreCase;
-        private static readonly TimeSpan RegexpSafetyTimeout = TimeSpan.FromSeconds(60);
-
-        private static readonly Regex RefinementSeparator = new("\\s+>\\s+", RegexpOpts, RegexpSafetyTimeout);
-        private static readonly Regex Hash = new("^hash:(?<num>\\d+)$", RegexpOpts, RegexpSafetyTimeout);
-        private static readonly Regex Verbatim = new("^@(?<verb>.*)$", RegexpOpts, RegexpSafetyTimeout);
-        private static readonly Regex IsRegex = new("^r:(?<pat>.*)$", RegexpOpts, RegexpSafetyTimeout);
         private static readonly Regex Whitespace = new("\\s+", RegexpOpts, RegexpSafetyTimeout);
         private static readonly Regex Or = new("\\|", RegexpOpts, RegexpSafetyTimeout);
         private static readonly Regex Negation = new("^(?'Open'\\(\\?:)*\\!(?<term>.+?)(?'Close-Open'\\))*$", RegexpOpts, RegexpSafetyTimeout);
         private static readonly Regex SquashExtraWilds = new("((\\(\\?:)?\\.\\*\\??\\)?){2,}", RegexpOpts, RegexpSafetyTimeout);
 
-        private static readonly Func<Term, Term> HonorFileExtension = (Term term) => term.Type == TermType.Verbatim ? term : term with { Pattern = Regex.Replace(term.Pattern, "(^|\\W)\\.(?<term>\\w+?)", "$1\\.${term}") };
-
-        private static readonly Func<Term, Term> DropUnnecessaryGlobStars = (Term term) => term;
-
-        private static readonly Func<Term, Term> LimitOrToOneTerm = (Term term) => term.Type == TermType.Verbatim ? term : Or.IsMatch(term.Pattern) ? term with { Pattern = $"(?:{term.Pattern})" } : term;
```

Comment:
These were really hard to read, dunno if the new editorconfig allows simpler formatting for multi-conditional expressions? (I simplified the expressions anyway in this case)

Reply:
yeah I changed the editorconfig so that it doesn't auto-change everything into one big line :)